service/frontend/overrides.go

```diff
@@ -0,0 +1,82 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package frontend
+
+import (
+	"context"
+
+	"github.com/blang/semver/v4"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/server/common/headers"
+)
+
+// Overrides defines a set of special case behaviors like compensating for buggy
+// SDK implementations
+type Overrides struct {
+	minTypeScriptEagerActivitySupportedVersion semver.Version
+}
+
+func NewOverrides() *Overrides {
+	return &Overrides{
+		minTypeScriptEagerActivitySupportedVersion: semver.MustParse("1.4.4"),
+	}
+}
+
+func (o *Overrides) shouldForceDisableEagerDispatch(sdkName, sdkVersion string) bool {
+	if sdkName == "temporal-python" && sdkVersion == "0.1b2" {
+		return true
+	} else if sdkName == "temporal-typescript" {
```

Comment:
nit: use constants defined in `headers`

Reply:
Yes, I wrote this before I realized we have those constants and forgot to edit.
Thanks.